docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+| `probe_override` | Dict[str, Any]               | None        | Override all or a portion of model's probed attributes |
+| `metadata`       | ModelSourceMetadata          | None        | Provide metadata that will be added to model's config |
+| `access_token`   | str                          | None        | Provide authorization information needed to download |
+| `priority`       | int                          | 10          | Download queue priority for the job       |
+
+
+The `inplace` field controls how local model Paths are handled. If
+True (the default), then the model is simply registered in its current
+location by the installer's `ModelConfigRecordService`. Otherwise, the
+model will be moved into the location specified by the  `models_dir`
+application configuration parameter. 
+
+The `variant` field is used for HuggingFace repo_ids only. If
+provided, the repo_id download handler will look for and download
+tensors files that follow the convention for the selected variant:
+
+- "fp16" will select files named "*model.fp16.{safetensors,bin}"
+- "onnx" will select files ending with the suffix ".onnx"
+- "openvino" will select files beginning with "openvino_model"
+
+In the special case of the "fp16" variant, the installer will select
+the 32-bit version of the files if the 16-bit version is unavailable.
+
+`subfolder` is used for HuggingFace repo_ids only. If provided, the
+model will be downloaded from the designated subfolder rather than the
+top-level repository folder. If a subfolder is attached to the repo_id
+using the format `repo_owner/repo_name:subfolder`, then the subfolder
+specified by the repo_id will override the subfolder argument.
+
+`probe_override` can be used to override all or a portion of the
+attributes returned by the model prober. This can be used to overcome
+cases in which automatic probing is unable to (correctly) determine
+the model's attribute. The most common situation is the
+`prediction_type` field for sd-2 (and rare sd-1) models. Here is an
+example of how it works:
+
+```
+install_job = installer.install_model(
+               source='stabilityai/stable-diffusion-2-1',
+			   variant='fp16',
+			   probe_override=dict(
+			         prediction_type=SchedulerPredictionType('v_prediction')
+	           )
+	      )
+```
+
+`metadata` allows you to attach custom metadata to the installed
+model. See the next section for details.
+
+`priority` and `access_token` are passed to the download queue and
+have the same effect as they do for the DownloadQueueServiceBase.
+
+#### Monitoring the install job process
+
+When you create an install job with `model_install()`, events will be
+passed to the list of `DownloadEventHandlers` provided at installer
+initialization time. Event handlers can also be added to individual
+model install jobs by calling their `add_handler()` method as
+described earlier for the `DownloadQueueService`.
```

Comment:
Agree. I think that updating the model config store is ok for now because it thread locks. There's no locking done when the model's being moved from the temporary directory to its final location, however. and I guess this could be a problem?

I'm wondering if the best way to implement this is to have the model installer class creates a queue for handling the completion routines? When a download is finished, the download thread will place a completion routine on the queue, and then a single worker thread would dequeue and execute it. This would force all completion routines to run in a single threaded manner.

Reply:
Good questions ... I'm going to have to give this some more thought. I feel like there's a way to simplify the download queue while achieving all of the same goals, just need to find some time to dig a little deeper. Shouldn't hold up the first PR to switch to using model IDs at least.